SPEC.md

```diff
@@ -7367,6 +7368,93 @@ Example output:
 </p>
 </details>
 
+### ✨ `join_paths`
+
+```
+File join_paths(File|String, String)
+File join_paths(File, Array[String]+)
+File join_paths(Array[String]+)
+```
+
+Joins together two or more paths into an absolute path in the host filesystem.
+
+There are three variants of this function:
+
+1. `File join_paths(File|String, String)`: Joins together exactly two paths. The first path may be either absolute or relative and must specify a directory; the second path is relative to the first path and may specify a file or directory.
+2. `File join_paths(File, Array[String]+)`: Joins together any number of relative paths with a base path. The first argument may be either an absolute or a relative path and must specify a directory. The paths in the second array argument must all be relative. The *last* element may specify a file or directory; all other elements must specify a directory.
+3. `File join_paths(Array[String]+)`: Joins together any number of paths. The array must not be empty. The *first* element of the array may be either absolute or relative; subsequent path(s) must be relative. The *last* element may specify a file or directory; all other elements must specify a directory.
+
+An absolute path starts with `/` and indicates that the path is relative to the root of the container in which the task is executed. A relative path does not start with `/` and indicates the path is relative to its parent directory. It is up to the execution engine to determine which directory to use as the parent when resolving relative paths; by default it is the working directory in which the task is executed.
```

Comment:
Do we want to prescribe the behaviour of supplying an absolute path as a later parameter, eg: Python will ignore all previous paths - or should this be an exception?

Reply:
I don't really like that behavior in python tbh. IMO if we're going to the principle of least surprising behavior, then that should be an error.